### PR TITLE
changed naming convention so latest file overwrites instead of adds t…

### PR DESCRIPTION
**Pull request by SteveScott**
…o latest directory.

#196 
Problem noted in issue #216

### CODE REVIEW

File: recipes/mastercard/runner.sh [hunk 1]
```diff
@@ -72,7 +90,8 @@ VERSION=$DATE
     )
     
     #unsplit csv is too large. Must compress.
-    zip -9 output/mastercard_$DATE.zip output/mastercard.csv
+    echo "compressing mastercard.csv"
+    zip -9 output/daily_transactions.zip output/mastercard.csv
```
Comment (YuliaZamriy) [inline recipes/mastercard/runner.sh:94]: does it need to be zipped in "output" folder?

File: recipes/mastercard/runner.sh [hunk 2]
```diff
@@ -34,20 +47,25 @@ VERSION=$DATE
...
-    echo "unzipping " $( ls -ltr input | tail -1 | awk '{print $NF}') 
+    echo 'listing...\n'
+    ls -ltr input
+    FULL_FILENAME=$( ls -ltr input | tail -1 | awk '{print $NF}') 
+    echo "unzipping " $FULL_FILENAME
     #unzips the first file by chronological order by sorting by modified date, reversed, and taking the tail to avoid the header
-    # (there is a check above to ensure there is only be one zip file so sorting is redundant).
     # Then use awk to select filename.
-    unzip -d /input -P $MASTERCARD_PASSWORD $( ls -ltr input | tail -1 | awk '{print $NF}') 
+    unzip -d /input -P $MASTERCARD_PASSWORD $FULL_FILENAME
```
Comment (YuliaZamriy) [inline recipes/mastercard/runner.sh:68]: where is $MASTERCARD_PASSWORD created? Is it in the .env file?

File: recipes/mastercard/runner.sh [hunk 3]
```diff
@@ -6,26 +6,39 @@ NAME=$(basename $BASEDIR)
...
     #was having trouble writing to input. Input directory is temporary and will not persist.
     chmod 777 input
     
-    #check to verify there is only one file on the mastercard server. 
-    #Files will not download and delete unless there is only one zip file. 
+    #check to verify there is a file on the mastercard server.
+    #Files will not download and delete unless there is at least one zip file. 
+    #decision to not check for single file.
     #comment out for texting
     ROWCOUNT=$(echo 'ls -l' | sftp -q -oPort=22022 -o StrictHostKeyChecking=no -i ~/.ssh/id_rsa_axway newyorkcity@files.mastercard.com:geoinsights/data/fromMC | grep .zip | wc -l)
```
Comment (YuliaZamriy) [inline recipes/mastercard/runner.sh:24]: what if there are multiple zip files? which row count is it going to capture?

File: recipes/mastercard/runner.sh [hunk 4]
```diff
@@ -82,4 +101,9 @@ VERSION=$DATE
     Upload $NAME latest
```
Comment (YuliaZamriy) [inline recipes/mastercard/runner.sh:101]: Is $NAME the same as "output/daily_transactions.zip"?

**Review by YuliaZamriy — COMMENTED** [review]
I have some questions before approving

File: recipes/mastercard/runner.sh (hunk 1 above)
Comment (SteveScott) [reply recipes/mastercard/runner.sh:94]: The mastercard csv is too large to upload. It must be compressed, and the raw csv deleted.

File: recipes/mastercard/runner.sh (hunk 2 above)
Comment (SteveScott) [reply recipes/mastercard/runner.sh:68]: Ah yes. I need to add this to the environment.

File: recipes/mastercard/runner.sh (hunk 3 above)
Comment (SteveScott) [reply recipes/mastercard/runner.sh:24]: the row count is the number of lines returned with the ls command. If there is one file, ROWCOUNT=1. Two files, ROWCOUNT=2.

File: recipes/mastercard/runner.sh (hunk 4 above)
Comment (SteveScott) [reply recipes/mastercard/runner.sh:101]: NAME=$(basename $BASEDIR). Name is probably "mastercard"
When one calls Upload, it uploads all files in the output directory. Upload is a user-created function.

File: recipes/mastercard/runner.sh (hunk 1 above)
Comment (YuliaZamriy) [reply recipes/mastercard/runner.sh:94]: I was asking more about zipping it in the output folder vs. zipping the file.

File: recipes/mastercard/runner.sh (hunk 4 above)
Comment (YuliaZamriy) [reply recipes/mastercard/runner.sh:101]: I see.

**Review by YuliaZamriy — APPROVED** [review]

**Pull request merged**